app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { createRecipeService, getRecipeByNameService } from 'components/RecipeManagement/services';
+import RecipeForm from 'components/RecipeManagement/RecipeForm';
+import {
+  ICreateRecipeProps,
+  IRecipeFormData,
+  IRecipeNameErrorData,
+} from 'components/RecipeManagement/types';
+import T from 'i18n-react';
+import { debounce } from 'lodash';
+import React, { FormEvent, useRef, useState, useReducer } from 'react';
+import { ActionType } from 'components/RecipeList/types';
+import {
+  reducer,
+  defaultInitialState,
+  Actions,
+  noErrorState,
+} from 'components/RecipeManagement/reducer';
+
+const PREFIX = 'features.WranglerNewUI.RecipeForm.labels';
+const recipeNameRegEx = /^[a-z\d\s]+$/i;
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [createRecipeState, dispatch] = useReducer(reducer, defaultInitialState);
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeFormData>({
+    recipeName: '',
+    description: '',
+  });
+
+  const [isSaveDisabled, setIsSaveDisabled] = useState<boolean>(true);
+
+  const setRecipeNameErrorData = (
+    recipeNameError: IRecipeNameErrorData,
+    formData: IRecipeFormData = recipeFormData
+  ) => {
+    dispatch({
+      type: Actions.SET_RECIPE_NAME_ERROR_STATE,
+      payload: recipeNameError,
+    });
+    handleSaveButtonMode(formData, recipeNameError);
+  };
+
+  /*
+  TODO: This static data has to be removed when we have actual API data,
+  then directly we will get that data from store as directives
+  */
+
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  const handleSaveButtonMode = (
+    formData: IRecipeFormData = recipeFormData,
+    nameErrorData: IRecipeNameErrorData = createRecipeState.recipeNameErrorData
+  ) => {
+    if (
+      formData.recipeName === '' ||
+      formData.description === '' ||
+      formData.recipeName?.trim().length === 0 ||
+      formData.description?.trim().length === 0 ||
+      nameErrorData.isRecipeNameError
+    ) {
+      setIsSaveDisabled(true);
+    } else {
+      setIsSaveDisabled(false);
+    }
+  };
+
+  const handleRecipeFormData = (formData: IRecipeFormData) => {
+    setRecipeFormData(formData);
+    handleSaveButtonMode(formData);
+  };
+
+  const onRecipeNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
why not debounce this whole function instead of just the bottom one? you're triggering a state change on every change

Reply:
instead of just validateIfRecipeNameExists*

Reply:
if we debounce the whole function the state will not updated properly as debounce will wait for some time to update so we are updating the state before the debounce function so that state will update properly